src/ra_log_wal.erl

```diff
@@ -527,12 +529,29 @@ open_wal(File, Max, #conf{write_strategy = o_sync,
         end;
 open_wal(File, Max, #conf{file_modes = Modes} = Conf) ->
     {ok, Fd} = ra_file_handle:open(File, Modes),
+    maybe_pre_allocate(Fd, Max),
     ok = write_header(Fd),
     ok = ra_file_handle:sync(Fd),
     {Conf, #wal{fd = Fd,
                 max_size = Max,
                 filename = File}}.
 
+maybe_pre_allocate(Fd, Max) ->
+    case ra_env:wal_pre_allocate() of
```

Comment:
This does a call into the configuration process, can we get this value at init time and stash it in the `conf` record?